cmd/ftl/cmd_dev.go

```diff
@@ -3,25 +3,45 @@ package main
 import (
 	"context"
 	"errors"
+	"fmt"
 	"time"
 
 	"golang.org/x/sync/errgroup"
 
+	"github.com/BurntSushi/toml"
 	"github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/ftlv1connect"
 	"github.com/TBD54566975/ftl/buildengine"
+	"github.com/TBD54566975/ftl/common/projectconfig"
 	"github.com/TBD54566975/ftl/internal/rpc"
 )
 
 type devCmd struct {
+	//cf.DefaultConfigMixin
 	Parallelism int           `short:"j" help:"Number of modules to build in parallel." default:"${numcpu}"`
-	Dirs        []string      `arg:"" help:"Base directories containing modules." type:"existingdir" required:""`
+	Dirs        []string      `arg:"" help:"Base directories containing modules." type:"existingdir" optional:""`
 	External    []string      `help:"Directories for libraries that require FTL module stubs." type:"existingdir" optional:""`
 	Watch       time.Duration `help:"Watch template directory at this frequency and regenerate on change." default:"500ms"`
 	NoServe     bool          `help:"Do not start the FTL server." default:"false"`
 	ServeCmd    serveCmd      `embed:""`
 }
 
 func (d *devCmd) Run(ctx context.Context) error {
+	if len(d.Dirs) == 0 && len(d.External) == 0 {
+		// TODO: is there a way to read this from ProjectConfigResolver?
```

Comment:
I couldn't figure out a good way of getting the project config, or reading out the directories from a config manager.